c++/src/kj/common.h

```diff
@@ -609,6 +610,52 @@ constexpr bool canMemcpy() {
   static_assert(kj::canMemcpy<T>(), "this code expects this type to be memcpy()-able");
 #endif
 
+namespace _ {  // private
+#if defined(__clang__) && defined(__is_identifier)
+#define KJ_HAS_BUILTIN_TYPE_TRAIT(x) (!__is_identifier(x))
+#elif defined(__has_builtin)
+#define KJ_HAS_BUILTIN_TYPE_TRAIT(x) __has_builtin(x)
+#else
+#define KJ_HAS_BUILTIN_TYPE_TRAIT(x) 0
+#endif
+
+#if KJ_HAS_BUILTIN_TYPE_TRAIT(__is_integral)
+template <typename T>
+constexpr bool canMemcmpArrayOf() {
+  return __is_integral(T);
```

Comment:
IMO it's not really worth doing all the checks to use __is_integral() if we still have to have the fallback for other compilers... I'd just use the fallback on all compilers. The benefit of __is_integral() would be if we could use it everywhere and avoid the added code complexity.